### PR TITLE
executor: set time zone offset in `tipb.SelectRequest`.

### PR DESCRIPTION
**Pull request by coocood**
The time zone offset will be used by TiKV to compare timestamp type.

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @tiancaiamao @zimulala @BusyJay PTAL

File: executor/executor_distsql.go [hunk 1]
```diff
@@ -806,3 +810,9 @@ func (e *XSelectTableExec) Next() (*Row, error) {
 func (e *XSelectTableExec) Fields() []*ast.ResultField {
 	return nil
 }
+
+// timeZoneOffset return local time zone offset in seconds.
```
Comment (zimulala) [inline executor/executor_distsql.go]: s/return/returns

File: executor/executor_distsql.go [hunk 2]
```diff
@@ -511,6 +512,7 @@ func (e *XSelectIndexExec) fetchHandles(idxResult distsql.SelectResult, ch chan<
 func (e *XSelectIndexExec) doIndexRequest() (distsql.SelectResult, error) {
 	selIdxReq := new(tipb.SelectRequest)
 	selIdxReq.StartTs = e.startTS
+	selIdxReq.TimeZoneOffset = proto.Int64(timeZoneOffset())
```
Comment (shenli) [inline executor/executor_distsql.go]: Why not use gogo nullable=false?

Comment (coocood) [inline executor/executor_distsql.go]: Forgot to do that.
We can fix this the next time we need to update tipb.

Comment (coocood) [inline executor/executor_distsql.go]: Forgot to do that.
We can fix this the next time we need to update tipb.

**Comment by shenli** [comment]
LGTM
/cc @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**